notebooks/post_DL3_analysis.ipynb

```diff
@@ -0,0 +1,2633 @@
+{
```

Comment:
<p>What is the interest of having energy_fit_edges? And why create it from a different MapAxis? </p><p>Is it possible (without strange behaviours) to have e_reco_m{in,ax} != e_fit_m{in,ax} ? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/cta-observatory/cta-lstchain/pull/955/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Reply:
<p>The energy_fit_edges can be different than energy_axis, in the sense of energy bins only. From gammapy dev - "Gammapy does not do unfolding of spectra (yet). It uses the 'hybrid' approach of computing flux points in reconstructed energy using forward folding and assuming flux points are not strongly correlated".</p><p>So for example, </p><p>if energy_axis.edges = [0.01, 0.1, 1, 10, 100] TeV</p><p>then we can have, </p><p>energy_fit edges.edges = [0.01, 0.1, 1, 10, 100] TeV, or </p><p>energy_fit edges.edges = [0.1, 1, 10, 100] TeV</p><p>And similarly, for LC estimation, we can use [0.1, 10] TeV</p><p>but not something like [0.5, 2] TeV</p><p> </p><p>I created a different MapAxis object for the energy_fit_edges, to have this distinction. But maybe, showing them as the same as energy_axis.edges is misleading.</p>

Reply:
Ok so if I understood correctly, energy_fit_edges should be a subset of  energy_axis.edges ?
In this case maybe define it as such.

Reply:
<p>Spectral Fit will be done in energy edges:\n", energy_fit_edges) </p><p> </p><p>This is wrong, you use this edges only for the flux point computation.</p>

Reply:
And even here the first point actually produced starts at 89 GeV because (i think) you are using dataset filtered by the safe mask anyway.

Reply:
Ok, I will rephrase it. But the flux point computation is indeed done in the provided bins. As the safe mask is applied, the dataset does not fill the energy bins below ~89 GeV, as can be seen in the `flux_points.to_table(...)` block, where those rows have nans for the flux estimation.

Reply:
"the flux point computation is indeed done in the provided bins" -> I would not say it is done here. The method tries to use the first points but we don't provide it the data so it gives NaN. This is independent of the content of your DL3 file.

With this being the case, it seems like this edges definition (subset of the edges used to bin the data, only used to create flux points, and even there actually compute a value only if the safe mask allows) is useless. You could simply run the flux point computation over all intervals and leave the selection to your safemask.